dask/array/utils.py

```diff
@@ -58,6 +58,8 @@ def assert_eq(a, b, check_shape=True, **kwargs):
         adt = a.dtype
         _check_dsk(a.dask)
         a = a.compute(get=get_sync)
+        if hasattr(a, 'todense'):
+            a = a.todense()
```

Comment:
Currently I have to remove the `__array__` protocol to convince numpy ufuncs to work properly on sparse arrays without converting them to numpy arrays.  This is the same tradeoff that `scipy.sparse` had to make:

```python
In [1]: import scipy.sparse

In [2]: scipy.sparse.eye(5)
Out[2]: 
<5x5 sparse matrix of type '<class 'numpy.float64'>'
	with 5 stored elements (1 diagonals) in DIAgonal format>

In [3]: import numpy as np

In [4]: np.array(scipy.sparse.eye(5))
Out[4]: 
array(<5x5 sparse matrix of type '<class 'numpy.float64'>'
	with 5 stored elements (1 diagonals) in DIAgonal format>, dtype=object)
```

I *think* that this will improve when `__numpy_ufunc__` becomes standard, but until then I think that we're stuck with a `todense` method.